cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/TypeResolver.kt

```diff
@@ -135,29 +137,17 @@ open class TypeResolver(ctx: TranslationContext) : ComponentPass(ctx) {
         return false
     }
 
-    private fun handleNode(node: Node?) {
-        if (node is RecordDeclaration) {
-            for (t in typeManager.firstOrderTypes) {
-                if (t.name == node.name && t is ObjectType) {
-                    // The node is the class of the type t
-                    t.recordDeclaration = node
-                }
-            }
-        } else if (node is ImportDeclaration) {
-            // Update the imports, as they might have changed because of symbols created by
-            // record/type inference
-            node.updateImportedSymbols()
-        }
-    }
-
     override fun cleanup() {
         // Nothing to do
     }
 
     /** Resolves all types in [TypeManager.firstOrderTypes] using [resolveType]. */
     fun resolveFirstOrderTypes() {
         for (type in typeManager.firstOrderTypes.sortedBy { it.name }) {
-            if (type is ObjectType && type.typeOrigin == Type.Origin.UNRESOLVED) {
+            if (
+                type is ObjectType && type.typeOrigin == Type.Origin.UNRESOLVED ||
```

Comment:
Didn't you miss brackets here around the `||` expression?

Reply:
In line 94, there's a similar check. Should this change also apply there?

Reply:
The brackets should not be necessary from a logic point since it’s evaluated left to right of view but I can add them for clarity 

I can adjust the check in line 94 as well although in practice we only use „guessed“ in the old
Java fronted and that does not have typedefs 

we could problably just remove „guessed“ and make it unresolved as well

Reply:
Ok... I thought you want to have `type is ObjectType && (type.typeOrigin == Type.Origin.UNRESOLVED || type.typeOrigin == Type.Origin.GUESSED)`. Are there no GUESSED types which are not ObjectTypes?

I thought the origin GUESSED was introduced just recently mainly to clarify some things?

Reply:
> Ok... I thought you want to have `type is ObjectType && (type.typeOrigin == Type.Origin.UNRESOLVED || type.typeOrigin == Type.Origin.GUESSED)`. Are there no GUESSED types which are not ObjectTypes?
> 
> I thought the origin GUESSED was introduced just recently mainly to clarify some things?

 No, the GUESSED enum is a left over that is only used in the old Java frontend. And all of them are object types

Reply:
Ok. And in contrast we can have unresolved non-object types?

Reply:
No, I don’t think so

Reply:
Then, the check is unnecessary, isn't it?